src/details/ArborX_DetailsLegacy.hpp

```diff
@@ -55,14 +55,18 @@ class LegacyValues
   size_type size() const { return Access::size(_primitives); }
 };
 
-template <typename Callback, typename Value>
+template <typename Callback>
 struct LegacyCallbackWrapper
 {
   Callback _callback;
 
-  template <typename Predicate>
+  LegacyCallbackWrapper(Callback const &callback)
+      : _callback(callback)
+  {}
```

Comment:
This guy is only needed for C++17
I would probably go for deduction guidelines because these would be easier to audit when we bump the minimum requirements